git-magic: add page (#10889)
tldr-pages/tldr: pages/common/git-magic.md

# git magic

> Automate add, commit, and push routines.
> Part of `git-extras`.
> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-magic>.

- Commit changes with a generated message:

`git magic`

> - Add untracked files and commit changes with a generated message:

Reviewer: ```suggestion
- [a]dd untracked files and commit changes with a generated message:
```

`[a]` (case-sensitive) is a command mnemonic. Learn more on our [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md).

Author: I can't find anything about it in the style guide but I'll keep that in mind for the future. Thanks

`git magic -a`

> - Commit changes with a custom message:

Reviewer: ```suggestion
- Commit changes with a custom [m]essage:
```

`git magic -m "{{custom_commit_message}}"`

> - Edit the commit message before committing:

Reviewer: ```suggestion
- [e]dit the commit [m]essage before committing:
```

`git magic -em "{{custom_commit_message}}"`

> - Commit changes and push to remote:

Reviewer: ```suggestion
- Commit changes and [p]ush to remote:
```

`git magic -p`

> - Commit changes with a force push to remote:

Reviewer: ```suggestion
- Commit changes with a [f]orce [p]ush to remote:
```
